src/client/gui/lib/vm_table/bulk_actions.dart

```diff
@@ -9,6 +9,72 @@ import '../providers.dart';
 import '../vm_action.dart';
 import 'vms.dart';
 
+class ZonesDropdown extends ConsumerWidget {
+  const ZonesDropdown({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    // Pre-fetch zones data
+    final zonesAsync = ref.watch(zonesProvider);
+    return PopupMenuButton(
+      tooltip: 'Change zone availability',
+      offset: const Offset(0, 40),
+      child: OutlinedButton(
+        onPressed: null,
+        child: const Text('Zones', style: TextStyle(color: Colors.black)),
+      ),
+      itemBuilder: (context) {
+        return [
+          PopupMenuItem(
+            enabled: false,
+            child: zonesAsync.when(
+              loading: () => const SizedBox(
+                width: 20,
+                height: 20,
+                child: CircularProgressIndicator(),
+              ),
+              error: (_, __) => const Text('Error loading zones'),
+              data: (zones) => Column(
+                mainAxisSize: MainAxisSize.min,
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  for (final zone in zones) ...[
+                    _ZoneToggleRow(zone.name),
+                    if (zone != zones.last) const SizedBox(height: 8),
+                  ],
+                ],
+              ),
+            ),
+          ),
+        ];
+      },
+    );
+  }
+}
+
+class _ZoneToggleRow extends ConsumerWidget {
```

Comment:
This is missing some parts of the design, namely the borders and the number of running VMs per zone.